pkg/context/httpcontext.go

```diff
@@ -206,10 +249,40 @@ func (ctx *httpContext) Span() tracing.Span {
 	return ctx.span
 }
 
+// Fill the empty tag. If all existing Tags are filled, create new one.
 func (ctx *httpContext) AddTag(tag string) {
 	ctx.tags = append(ctx.tags, tag)
 }
 
+// Add new LazyTag.
+func (ctx *httpContext) AddLazyTag(ns string, prefix string, msg string, intMsg int) {
+	lazyTag := lazyTagPool.Get().(*LazyTag)
+	lazyTag.Namespace = ns
+	lazyTag.Prefix = prefix
+	lazyTag.StringMsg = msg
+	lazyTag.IntMsg = intMsg
+	lazyTag.Sep = ": "
+	ctx.lazyTags = append(ctx.lazyTags, lazyTag)
+}
+
+func (ctx *httpContext) ReleaseLazyTags() {
+	for _, lazyTag := range ctx.lazyTags {
+		lazyTagPool.Put(lazyTag)
+	}
+}
+
+// Return all tags in string format.
+func (ctx *httpContext) GetTags() []string {
```

Comment:
```
type Tag struct {
  tag string // used for add tag
  lazyTag func() string // used for lazy tag
}
```
is this a good idea? we combine tag and lazy tag together, so the order will not change...

Reply:
This is a good idea. Even simpler is to use `func() string` type for the slice, so we don't need struct at all:

```go
// Add new Tag.
func (ctx *httpContext) AddTag(tag string) {
	ctx.lazyTags = append(ctx.lazyTags, func() string {
		return tag
	})
}

// Add new LazyTag.
func (ctx *httpContext) AddLazyTag(lazyTag LazyTagFunc) {
	ctx.lazyTags = append(ctx.lazyTags, lazyTag)
}
```